opentelemetry-sdk/src/opentelemetry/sdk/logs/export/__init__.py

```diff
@@ -51,3 +58,215 @@ def shutdown(self):
 
         Called when the SDK is shut down.
         """
+
+
+class SimpleLogProcessor(LogProcessor):
+    """This is an implementation of LogProcessor which passes
+    received logs in the export-friendly LogData representation to the
+    configured LogExporter, as soon as they are emitted.
+    """
+
+    def __init__(self, exporter: LogExporter):
+        self._exporter = exporter
+        self._shutdown = False
+
+    def emit(self, log_data: LogData):
+        if self._shutdown:
+            _logger.warning("Processor is already shutdown, ignoring call")
+            return
+        token = attach(set_value("suppress_instrumentation", True))
+        try:
+            self._exporter.export((log_data,))
+        except Exception:  # pylint: disable=broad-except
+            _logger.exception("Exception while exporting logs.")
+        detach(token)
+
+    def shutdown(self):
+        self._shutdown = True
+        self._exporter.shutdown()
+
+    def force_flush(
+        self, timeout_millis: int = 30000
+    ) -> bool:  # pylint: disable=no-self-use
+        return True
+
+
+class _FlushRequest:
+    __slots__ = ["event", "num_log_records"]
+
+    def __init__(self):
+        self.event = threading.Event()
+        self.num_log_records = 0
+
+
+class BatchLogProcessor(LogProcessor):
```

Comment:
I don't think I quite understand the logic flow. How do the handlers (that pick up logging messages from the Python logger) communicate with the `Processor`? Are handlers added to the root log handler?

Reply:
> Are handlers added to the root log handler?

Correct. End-user will have to attach the OTLPHandler to the root logger.

> How do the handlers (that pick up logging messages from the Python logger) communicate with the Processor

Every log statement from the logging lib will be propagated to the root logger. Users can attach any number of Handlers with a logger instance. They may already be using out-of-box handlers from stdlib. `OTLPHandler` is such which sends the log data in OTLP data model format to configured destination (could be n/w destination or file etc...).

The logic flow is `Logger -> Handler -> LogEmitter -> LogProcessor -> LogExporter`. Think of Handler as Instrumentation and LogEmitter as Tracer from tracing SDK (kind of inaccurate but helps in visualizing things). Does it make sense now?

Reply:
Ahh I see, each handler SHOULD have a reference to a log emitter. We would expect users to possible create their own `Handlers` correct? Maybe it would be good to have another abstraction layer on top of `logging.Handler` specific for OpenTelemetry log handlers (similar to `BaseInstrumentor`). Thoughts?

Reply:
> We would expect users to possible create their own Handlers correct?

They don't need another handler for the same logging lib but they can create Handler like hooks for other third-party logging libs. 

>Maybe it would be good to have another abstraction layer on top of logging.Handler specific for OpenTelemetry log handlers (similar to BaseInstrumentor).

Yeah, thought about it but couldn't come up with anything meaningful. Do you have any rough idea in your head how you would like it to be?

Reply:
Nothing big for now, but similarly to instrumentations, `BaseInstrumentor` was needed eventually as we found more commonalities and functions between instrumentations. If a custom `Handler` is created for OpenTelemetry, maybe we need to it have an `emitter`?

Reply:
Also probably not blocking the PR, just something to note if we ever get confusion on HOW a handler should look like.